### PR TITLE
fix: back button on last 2 consignments screens

### PR DESCRIPTION
**Pull request by admbtlr**
The type of this PR is: Bugfix

This PR resolves [CX-1116]

### Description

Overrides the back button on the last two screens of the consignments form, to avoid resubmissions. If submission is still in progress, pressing back shows an alert and does nothing. If the submission is complete, pressing back exits the consignment form completely.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-1116]: https://artsyproduct.atlassian.net/browse/CX-1116

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The CHANGELOG is not valid YML:
YAMLSemanticError: Implicit map keys need to be followed by map values
    at resolveBlockMapItems (/home/circleci/project/node_modules/yaml/dist/schema/parseMap.js:225:31)
    at Object.parseMap [as resolve] (/home/circleci/project/node_modules/yaml/dist/schema/parseMap.js:33:79)
    at Schema.resolveNode (/home/circleci/project/node_modules/yaml/dist/schema/index.js:177:27)
    at Schema.resolveNodeWithFallback (/home/circleci/project/node_modules/yaml/dist/schema/index.js:200:22)
    at Document.resolveNode (/home/circleci/project/node_modules/yaml/dist/Document.js:494:22)
    at resolveBlockSeqItems (/home/circleci/project/node_modules/yaml/dist/schema/parseSeq.js:64:24)
    at Object.parseSeq [as resolve] (/home/circleci/project/node_modules/yaml/dist/schema/parseSeq.js:27:79)
    at Schema.resolveNode (/home/circleci/project/node_modules/yaml/dist/schema/index.js:177:27)
    at Schema.resolveNodeWithFallback (/home/circleci/project/node_modules/yaml/dist/schema/index.js:200:22)
    at Document.resolveNode (/home/circleci/project/node_modules/yaml/dist/Document.js:494:22)</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Upcoming an array in the CHANGELOG</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: CHANGELOG.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ upcoming:
   dev:
     - Improve relay environment setup - david
   user_facing:
-    - 
+    - Override Android back button in consignment form - adam
```
Comment (ds300) [inline CHANGELOG.yml:7]: These user facing entries end up in the iOS public changelog, so we should not include stuff related to wip android development yet. We need to figure out a new way of managing changelogs soon 😬 for now we can just put stuff in the dev section.

File: src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx [hunk 2]
```diff
@@ -32,6 +34,25 @@ const ConfirmContactInfo: React.FC<{
     setPhoneNumber(me?.phone)
   }, [me?.phone])
 
+  useFocusEffect(
```
Comment (ds300) [inline src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx:37]: 👍🏼 good thinking

File: src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx [hunk 3]
```diff
@@ -32,6 +34,25 @@ const ConfirmContactInfo: React.FC<{
...
+    React.useCallback(() => {
+      const onBackPress = () => {
+        if (!submissionRequestValidationCheck()) {
+          Alert.alert("Consignment Submission In Progress", "Please wait until your consignment has been submitted.", [
+            { text: "OK" },
+          ])
+        } else {
+          dismissModal()
+        }
```
Comment (ds300) [inline src/lib/Scenes/Consignments/Screens/ConfirmContactInfo.tsx:46]: hmm not sure about this here or on the confirmation screen. Seems like if something goes wrong and the `submissionRequestValidationCheck` never returns `true` then the user will be stuck on the confirm page unless they quit the app. If that's true I think an interrogative alert with copy like `Are you sure you want to quit? / Consignment submission in progress. / quit / keep editing` might be better.

**Review by ds300 — COMMENTED** [review]
Looks great! I just have a couple of minor concerns before we ship it.

**Review by ds300 — APPROVED** [review]
🙌🏼 thanks for fixing those alerts

**Pull request merged**